pkg/webhook/tenantresource/objects.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020-2021 Clastix Labs
+// SPDX-License-Identifier: Apache-2.0
+
+package tenant
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	capsulev1beta2 "github.com/clastix/capsule/api/v1beta2"
+	"github.com/clastix/capsule/controllers/resources"
+	capsulewebhook "github.com/clastix/capsule/pkg/webhook"
+	"github.com/clastix/capsule/pkg/webhook/utils"
+)
+
+type cordoningHandler struct{}
+
+func WriteOpsHandler() capsulewebhook.Handler {
+	return &cordoningHandler{}
+}
+
+func (h *cordoningHandler) handler(ctx context.Context, clt client.Client, decoder *admission.Decoder, req admission.Request, recorder record.EventRecorder) *admission.Response {
+	tntList := &capsulev1beta2.TenantList{}
+
+	if err := clt.List(ctx, tntList, client.MatchingFieldsSelector{Selector: fields.OneTermEqualSelector(".status.namespaces", req.Namespace)}); err != nil {
+		return utils.ErroredResponse(err)
+	}
+
+	obj := &unstructured.Unstructured{}
+	if err := decoder.Decode(req, obj); err != nil {
+		return utils.ErroredResponse(err)
+	}
+	// If the object is not managed by a TenantResource thanks to its label, ca ne skipped
+	if labels := obj.GetLabels(); labels != nil {
+		if _, ok := labels[resources.Label]; !ok {
+			return nil
+		}
+	}
```

Comment:
I'm still not convinced by this approach: I'd like to give it a try to the indexer since otherwise a Tenant Owner could lock a resource object by just using the label key.